manifests/init.pp

```diff
@@ -0,0 +1,166 @@
+  Integer[0,1]                $skip_inode_check                  = 0,
+  Optional[String[1]]         $ssh_config_dir                    = undef,
+  Optional[Array[String[1]]]  $startup_paths                     = undef, #suggestion: /etc/rc.d /etc/rc.local ??
+  Optional[String[1]]         $stat_cmd                          = undef,
+  Optional[Array[String[1]]]  $suspscan_dirs                     = undef,
+  Integer                     $suspscan_maxsize                  = 1024000,
+  String[1]                   $suspscan_temp                     = '/dev/shm',
+  Integer                     $suspscan_thresh                   = 200,
+  Optional[Array[String[1]]]  $suspscan_whitelist                = undef,
+  Optional[Array[String[1]]]  $syslog_config_file                = undef,
+  Optional[String[1]]         $tmpdir                            = undef,
+  Optional[Array[String[1]]]  $uid0_accounts                     = undef,
+  Optional[Array[String[1]]]  $update_lang                       = undef,
+  Integer[0,1]                $update_mirrors                    = 1,
+  Integer[0,1]                $updt_on_os_change                 = 0,
+  Integer[0,1]                $use_locking                       = 1,
+  Optional[Array[String[1]]]  $user_fileprop_files_dirs          = undef,
+  Integer[0,1]                $use_sunsum                        = 0,
+  String[1]                   $use_syslog                        = 'NONE', #recommend authpriv.notice??
+  Integer[0,1]                $warn_on_os_change                 = 1,
+  Optional[String[1]]         $web_cmd                           = undef, #shuold keep? rkhunter auto selects one
+  Integer[0,1]                $whitelisted_is_white              = 0,
+  Optional[Array[String[1]]]  $writewhitelist                    = undef,
+  Optional[Array[String[1]]]  $xinetd_allowed_svc                = undef,
+  Optional[String[1]]         $xinetd_conf_path                  = undef
+) {
+
+  package { 'unhide':
+    ensure => $package_ensure
+  }
+
+  package { 'rkhunter':
+    ensure  => $package_ensure,
+    require => Package['unhide']
+  }
+
+  file { $rkhunter_conf_file:
+    ensure  => 'file',
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0640',
+    path    => '/etc/rkhunter.conf',
+    content => template($rkhunter_conf_file_template),
+    require => Package['rkhunter']
+  }
+
+  if $check_for_updates {
+    cron { 'rkhunter_update':
+      command  => 'rkhunter --update --nocolors',
+      minute   => $minute_upd,
+      hour     => $hour_upd,
+      month    => $month_upd,
+      monthday => $monthday_upd,
+      weekday  => $weekday_upd,
+      require  => [ Package['rkhunter'], Package['unhide'], File[ $rkhunter_conf_file ] ]
+    }
+  }
+
+  cron { 'rkhunter':
+    command  => 'if test -f /var/lib/rkhunter/db/rkhunter.dat; then rkhunter --check --skip-keypress --quiet; else rkhunter --check --skip-keypress --quiet --propupd; fi',
```

Comment:
This should not run `propupd` on a regular basis.

See the `update_aide_db` `Exec` in the [aide module](https://github.com/simp/pupmod-simp-aide) for a potential way to do this.

Also, make running `propupd` optional (it really should be optional in the AIDE module as well).

Reply:
It is not being run on any basis. It HAS to be run the first time after installation to make sure you have an actual database else rkhunter has nothing to run against. This is a check to see if the system it is running on has a database in place, and if not it creates said database.

> Before running RKH you will need to fill the file properties database by
> running the following command:
> 
>     rkhunter --propupd
(excerpt from [rkhunter documentation](https://sourceforge.net/p/rkhunter/rkh_code/ci/master/tree/files/README))

Reply:
Outside of that the user is going to have to read the rkhunter documentation if they want to be changing parameters that might require a database update.

Reply:
@sharkbruhaha Reading through this, I'm assuming that `propupd` needs to be run if, and only if, `/var/lib/rkhunter/db/rkhunter.dat` does not exist.

Second, you allow the configuration of `dbdir` via the `rkhunter::config::dbdir` parameter.

Last, there are various parameters (which are documented) that require `propupd` to be run again.

If this is all true, I believe that the following needs to happen:

1. A class needs to be created that houses an exec that calls `rkhunter --propupd`. Something like `rkhunter::propupd` makes sense to me. I suppose that you could give it an `enable` parameter just in case the user wants to ensure that it's never triggered if the DB already exists. It should always run if the DB doesn't exist since `rkhunter` is fairly useless otherwise.
2. A fact needs to be created (I would make it a Hash called `rkhunter`) that indicates whether or not the database file exists. The fact will need to infer the location from the `rkhunter.conf` configuration file.
3. The class that was created will need to be notified if the DB does not exist.
4. The `cron` job should never run `--propupd`
5. I wouldn't bother checking to see if the DB file exists prior to running the cron job. If it fails, then something went wrong and the admin needs to check on the system.

Other Stuff:

* Booleans should almost always have a `true` or `false` default. There is usually no use to leave them `undef` unless they directly feed into a configuration file.
* Parameters appear to need realignment
* Ideally, we would trigger the `exec` class on updates to parameters that need it but that's a non-breaking change that can wait.

Reply:
made new ticket to take care of propupd work